cmd/mdatagen/cmd/mdatagen/internal/testhelpers/context_check_consumer.go

```diff
@@ -0,0 +1,46 @@
+package testhelpers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	TestKey   = "ctx-prop-key"
+	TestValue = "expected-value"
+)
+
+type ContextCheckLogsConsumer struct {
+	T *testing.T
+}
+
+func (c *ContextCheckLogsConsumer) ConsumeLogs(ctx context.Context, logs plog.Logs) error {
```

Comment:
This way of writing tests seems weird, and not really a Go pattern. Why can't we run the assertion inline?

Reply:
Thanks for the feedback. You're right that the current structure isn’t a typical Go pattern.
I introduced `ContextCheck*Consumer` as a lightweight reusable helper to avoid duplicating the same `context `assertion logic in each test case. The idea was to keep individual test methods focused on wiring rather than repeating assertions.
That said, I can definitely refactor it to assert inline or wrap the consumer in a more idiomatic test helper if preferred. Let me know if you’d like to see it changed.